sonaranalyzer-dotnet/src/SonarAnalyzer.Utilities/Rules.Description/S4226.html

```diff
@@ -0,0 +1,39 @@
+<p>It makes little sense to create an extension method when it is possible to just add that method to the type itself.</p>
```

Comment:
This justification doesn't apply if (1) the type is defined in a different assembly, or (2) the type is an interface.
I'm not saying that the rule shouldn't fire in these cases, just that the current wording will make it look like a false positive to the user.

Reply:
Extension methods can only be declared in top-level classes (had do double-check, compiler reports error CS1109).